homeassistant/components/device_tracker/xfinity.py

```diff
@@ -0,0 +1,59 @@
+"""Support for device tracking via Xfinity Gateways."""
+import logging
+
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import (
+    DOMAIN, PLATFORM_SCHEMA, DeviceScanner)
+from homeassistant.const import CONF_HOST
+
+REQUIREMENTS = ['xfinity-gateway==0.0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HOST = '10.0.0.1'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string
+})
+
+
+def get_scanner(hass, config):
+    """Validate the configuration and return an Xfinity Gateway scanner."""
+    scanner = XfinityDeviceScanner(config[DOMAIN][CONF_HOST])
+
+    return scanner if scanner.success_init else None
+
+
+class XfinityDeviceScanner(DeviceScanner):
+    """This class queries an Xfinity Gateway."""
+
+    def __init__(self, address):
+        """Initialize the scanner."""
+        from xfinity_gateway import XfinityGateway
+
+        try:
+            _LOGGER.debug('Initializing')
+            self.gateway = XfinityGateway(address)
```

Comment:
We can move the instantiation of the gateway to `get_scanner` and pass the gateway to `XfinityDeviceScanner` instead.